devicetypes/smartthings/zwave-multi-button.src/zwave-multi-button.groovy

```diff
@@ -0,0 +1,192 @@
+			state "battery", label:'${currentValue}% battery', unit:""
+		}
+
+		main "button"
+		details(["button", "battery"])
+	}
+}
+
+def installed() {
+	runIn(2, "initialize", [overwrite: true])
+	sendEvent(name: "button", value: "pushed", isStateChange: true)
+}
+
+def updated() {
+	runIn(2, "initialize", [overwrite: true])
+}
+
+
+def initialize() {
+	def numberOfButtons = prodNumberOfButtons[zwaveInfo.prod]
+	sendEvent(name: "numberOfButtons", value: numberOfButtons, displayed: false)
+	sendEvent(name: "checkInterval", value: 8 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	if(!childDevices) {
+		addChildButtons(numberOfButtons)
+	}
+	if(childDevices) {
+		def event
+		for(def endpoint : 2..prodNumberOfButtons[zwaveInfo.prod]) {
+			event = createEvent(name: "button", value: "pushed", isStateChange: true)
+			sendEventToChild(endpoint, event)
+		}
+	}
+	response([
+			secure(zwave.batteryV1.batteryGet()),
+			"delay 2000",
+			secure(zwave.wakeUpV1.wakeUpNoMoreInformation())
+	])
+}
+
+def configure() {
+	def cmds = []
+	for(def parameter : 21..26) {
+		cmds += secure(zwave.configurationV1.configurationSet(parameterNumber: parameter, scaledConfigurationValue: 15)) //Makes Fibaro KeyFob buttons send all kind of supported events
+	}
+	cmds
+}
+
+def parse(String description) {
+	def result = []
+	if (description.startsWith("Err")) {
+		result = createEvent(descriptionText:description, isStateChange:true)
+	} else {
+		def cmd = zwave.parse(description)
+		if (cmd) {
+			result += zwaveEvent(cmd)
+		}
+	}
+	log.debug "Parse returned: ${result}"
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.centralscenev1.CentralSceneNotification cmd) {
+	def value = eventsMap[(int) cmd.keyAttributes]
+	def description = "Button no. ${cmd.sceneNumber} was ${value}"
+	def event = createEvent(name: "button", value: value, descriptionText: description, data: [buttonNumber: cmd.sceneNumber], isStateChange: true)
+	if(cmd.sceneNumber == 1) {
+		return event
+	} else {
+		sendEventToChild(cmd.sceneNumber, event)
+		return createEvent(descriptionText: description)
+	}
+}
+
+def sendEventToChild(buttonNumber, event) {
+	String childDni = "${device.deviceNetworkId}:$buttonNumber"
+	def child = childDevices.find { it.deviceNetworkId == childDni }
+	child?.sendEvent(event)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv1.WakeUpNotification cmd) {
+	def results = []
+	results += createEvent(descriptionText: "$device.displayName woke up", isStateChange: false)
+	if (!state.lastbatt || (now() - state.lastbatt) >= 56*60*60*1000) {
```

Comment:
I think from this point onward we just want to poll the battery every time we get a wakeup. It's a new certification requirement that we'll have to go back and re-do for all old DTHs, so we should just start doing it now.